src/ape/api/transactions.py

```diff
@@ -233,14 +237,34 @@ def decode_logs(
         Returns:
             Iterator[:class:`~ape.types.ContractLog`]
         """
-        if not isinstance(abi, (list, tuple)):
-            abi = [abi]
+        if abi:
+            if not isinstance(abi, (list, tuple)):
+                abi = [abi]
 
-        for event_abi in abi:
-            if not isinstance(event_abi, EventABI):
-                event_abi = event_abi.abi
+            event_abis: List[EventABI] = [a.abi if not isinstance(a, EventABI) else a for a in abi]
+            yield from self.provider.network.ecosystem.decode_logs(event_abis, self.logs)
 
-            yield from self.provider.network.ecosystem.decode_logs(event_abi, self.logs)
+        else:
+            # If ABI is not provided, decode all events
+            addresses = {x["address"] for x in self.logs}
+            contract_types = self.chain_manager.contracts.get_multiple(addresses)
+            # address → selector → abi
+            selectors = {
+                address: {keccak(text=abi.selector): abi for abi in contract.events}
```

Comment:
`keccak` makes it ethereum-specific, can this fall through into an `EcosystemAPI` function (maybe `decode_all_logs(...)`)